components/script/dom/webglrenderingcontext.rs

```diff
@@ -1077,6 +1077,36 @@ impl WebGLRenderingContextMethods for WebGLRenderingContext {
             .unwrap()
     }
 
+    #[allow(unsafe_code)]
+    // https://www.khronos.org/registry/webgl/specs/latest/1.0/#5.14.12
+    fn ReadPixels(&self, _cx: *mut JSContext, x: i32, y: i32, width: i32, height: i32,
+                  format: u32, pixel_type: u32, pixels: *mut JSObject) {
+        let mut data = match unsafe { array_buffer_view_data::<u8>(pixels) } {
+            Some(data) => data,
+            None => return self.webgl_error(InvalidValue),
+        };
+
+        match unsafe { JS_GetArrayBufferViewType(pixels) } {
+            Type::Uint8 => (),
+            _ => return self.webgl_error(InvalidOperation)
+        }
+
+        let (sender, receiver) = ipc::channel().unwrap();
+        self.ipc_renderer
+            .send(CanvasMsg::WebGL(WebGLCommand::ReadPixels(x, y, width, height, format, pixel_type, sender)))
+            .unwrap();
+
+        let result = receiver.recv().unwrap();
+
+        if result.len() > data.len() {
+            return self.webgl_error(InvalidOperation)
+        }
+
+        for i in 0..result.len() {
```

Comment:
nit: this could use [copy_from_slice](https://doc.rust-lang.org/std/primitive.slice.html#method.copy_from_slice)